python/tank_vendor/shotgun_authentication/console_authentication.py

```diff
@@ -101,20 +119,17 @@ def _get_keyboard_input(self, label, default_value=""):
             user_input = raw_input(text) or default_value
         return user_input
 
-    def _get_session_token(self, hostname, login, password, http_proxy):
+    def _get_2fa_code(self):
         """
-        Retrieves a session token for the given credentials. If it fails, the user is informed
-        :param hostname: The host to connect to.
-        :param login: The user to get a session for.
-        :param password: Password for the user.
-        :param http_proxy: Proxy to use. Can be None.
-        :returns: If the credentials were valid, returns a session token, otherwise returns None.
+        Prompts the user for his 2fa code.
+        :returns: Two factor authentication code.
+        :raises AuthenticationCancelled: If the user enters an empty code, the exception will be
+                                         thrown.
         """
-        try:
-            return session_cache.generate_session_token(hostname, login, password, http_proxy)
-        except AuthenticationError:
-            print "Login failed."
-            return None
+        code = raw_input("Code (empty to abort): ")
```

Comment:
What does the full authentication interaction look like here? Wondering if it may be confusing for the user to suddenly get prompted:

```
Code (empty to abort):
```

Does it say something else? How about following the UI and instead say:

```
Shotgun two factor authentication - Please enter the code 
you received in the Google Authenticator app (empty to abort): 
```